### PR TITLE
CP-1415 Allow use of react-dart 0.9.0, w_flux

### PR DESCRIPTION
**Pull request by maxwellpeterson-wf**
## Issue
react-dart 0.9.0 is available.

## Changes
**Source:**
- Update react dependency range.

**Tests:**
- Fix tests that are currently broken in master.

## Areas of Regression
- Examples

## Testing
- CI passes
- Examples function correctly

## Code Review
@trentgrover-wf
@evanweible-wf 
@dustinlessard-wf
FYI @jayudey-wf
FYI @aaronlademann-wf

### CODE REVIEW

File: test/store_test.dart [hunk 1]
```diff
@@ -111,3 +114,7 @@ void main() {
     });
   });
 }
+
+Future nextTick() {
+  return new Future.delayed(new Duration());
```
Comment (evanweible-wf) [inline test/store_test.dart]: deja vu..

File: test/action_test.dart [hunk 2]
```diff
@@ -35,29 +35,32 @@ void main() {
...
-    test('should support dispatch without a payload', () {
+    test('should support dispatch without a payload', () async {
       Action _action = new Action();
 
       _action.listen((payload) => expectAsync((payload) {
             expect(payload, equals(null));
           }));
 
+      await nextTick();
       _action();
```
Comment (maxwellpeterson-wf) [inline test/action_test.dart]: Tried it, doesn't work :/

File: test/store_test.dart (hunk 1 above)
Comment (maxwellpeterson-wf) [inline test/store_test.dart]: Fixed

File: test/action_test.dart (hunk 2 above)
Comment (jayudey-wf) [inline test/action_test.dart]: double :/

File: test/component_test.dart [hunk 3]
```diff
@@ -21,11 +21,9 @@ import 'package:react/react.dart' as react;
 import 'package:test/test.dart';
 import 'package:w_flux/w_flux.dart';
 
-void main() {
-  Future nextTick() {
-    return new Future.delayed(new Duration(milliseconds: 1));
-  }
+import 'utils.dart';
```
Comment (jayudey-wf) [inline test/component_test.dart]: is this import needed anymore?

Comment (maxwellpeterson-wf) [inline test/component_test.dart]: Yeah, component_test.dart was already using a method like this.

File: test/action_test.dart [hunk 4]
```diff
@@ -35,30 +35,39 @@ void main() {
...
-      _action.listen((payload) => expectAsync((payload) {
-            expect(payload, equals(null));
-          }));
+      _action.listen((String payload) {
+        expect(payload, equals(null));
+        c.complete();
+      });
 
       _action();
+      return c.future;
```
Comment (jayudey-wf) [inline test/action_test.dart]: does it make a difference as to whether you are returning the future rather than awaiting it?

Comment (maxwellpeterson-wf) [inline test/action_test.dart]: I don't think so. @evanweible-wf or @trentgrover-wf ?

Comment (evanweible-wf) [inline test/action_test.dart]: no difference

**Comment by evanweible-wf** [comment]
+1

**Comment by jayudey-wf** [comment]
+1

**Comment by trentgrover-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
@jayudey-wf what's the issue with the test?

**Comment by jayudey-wf** [comment]
the two test bodies look to have the same content

```
    test('should support dispatch with a payload', () async {
      Completer c = new Completer();
      action.listen((String payload) {
        expect(payload, equals('990 guerrero'));
        c.complete();
      });

      action('990 guerrero');
      return c.future;
    });

    test('should dispatch by default when called', () async {
      Completer c = new Completer();
      action.listen((String payload) {
        expect(payload, equals('990 guerrero'));
        c.complete();
      });

      action('990 guerrero');
      return c.future;
    });
```

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - passing CI, examples working as expected
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Pull request merged**